dagmc_stats/DagmcQuery.py

```diff
@@ -43,3 +53,29 @@ def get_tris(self):
             tris = self.dagmc_file._my_moab_core.get_entities_by_type(item, types.MBTRI)
             tris_lst.extend(tris)
         return tris
+
+    def calc_tris_per_vert(self, ignore_zero=True):
+        """
+        calculate triangle per vertex data
+        
+        inputs
+        ------
+        ignore_zero : (boolean) whether or not to ignore zero tris_per_vert values
+
+        outputs
+        -------
+        none
+        """
+        t_p_v_data = []
+        tri_dimension = 2
+        for vertex in self.dagmc_file._my_moab_core.get_entities_by_type(
+                self.meshset, types.MBVERTEX):
+            tpv_val = self.dagmc_file._my_moab_core.get_adjacencies(vertex, tri_dimension).size()
+            if ignore_zero and tpv_val == 0:
+                continue
+            row_data = {'vert_eh': vertex, 'tri_per_vert': tpv_val}
+            t_p_v_data.append(row_data)
+        if self._vert_data.empty:
+            self._vert_data = self._vert_data.append(t_p_v_data)
+        else:
```

Comment:
If each query is associated with a single meshset, is it possible for this to be non-empty?

Reply:
It's checking if the whole dataframe is empty. It is possible to call multiple queries that are indexed by vertex on a single meshset such as this tris per vert or roughness that would all go into this same dataframe. If one is called before the other, the dataframe may not be empty.